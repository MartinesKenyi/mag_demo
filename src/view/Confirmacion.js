import React from 'react';

export const Confirmacion = () => {

    const dimensiones = window.matchMedia("(max-width: 580px)").matches;

    return (
        <div className="background__body">
            <div className="filter"
                style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}
            >
                <div className="confirmacion__container">

                    <div className="row">
                        <div style={ (dimensiones) ? {width: '100%', float: 'left'} : { width: '65%', float: 'left' }}>

                            <div style={{ padding: '30px' }}>

                                <div className="success-checkmark">
                                    <div className="check-icon">
                                        <span className="icon-line line-tip"></span>
                                        <span className="icon-line line-long"></span>
                                        {/* <div className="icon-circle"></div> */}
                                        <div className="icon-fix"></div>
                                    </div>
                                </div>

                                <div style={{ textAlign: 'center' }}>
                                    <p className="confirmacion__bienvenido"> Bienvenido,
                                    <span> NOMBRE </span>
                                    has adquirido el
                                </p>
                                    <h2 className="confirmacion__plan">Plan Est√°ndar</h2>
                                    <span className="confirmacion__precio">S/29 al mes</span>

                                    <p className="confirmacion__descripcion">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>

                                    <div className="confirmacion__ref">
                                        Ir a ver mi plan
                                </div>
                                </div>
                            </div>
                        </div>
                        <div style={{ width: '35%', float: 'left' }} hidden={dimensiones}>
                            <div>
                                <img
                                    src={require('../assets/img/success3.jpg').default}
                                    alt="img"
                                    className="confirmacion__img"
                                />
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
