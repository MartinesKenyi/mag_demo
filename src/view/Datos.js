import React, { useState, useContext } from 'react';
import { Redirect } from 'react-router'
import { FormDatos } from '../components/FormDatos';
import { SelectPlans } from '../components/SelectPlans';
import { SuscriptionContext } from '../suscription/SuscriptionContext';

export const Datos = ({ history }) => {

    const dimensiones = window.matchMedia("(max-width: 650px)").matches;
    const [modal, setModal] = useState(true);
    const { suscriptionState } = useContext(SuscriptionContext);

    const { ok } = suscriptionState
    const { popular } = suscriptionState.sucriptionActive;

    if(!ok) {
        return <Redirect to="/suscripcion" />
    }

    const handleModal = () => {
        setModal(!modal)
    }

    return (
        <div className="background__body">
            <div className="filter"
                style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}
            >
                <div className="datos__container">
                    <div className="row">
                        <div style={{ width: '40%', float: 'left' }} hidden={dimensiones}>
                            <SelectPlans />
                        </div>
                        <div style={(dimensiones) ? { width: '100%', float: 'left' } : { width: '60%', float: 'left' }}>
                            <FormDatos
                                history={history}
                            />
                        </div>

                        <div 
                            className="datos__cambiar" 
                            hidden={!dimensiones}
                            onClick={handleModal}
                        >
                            {(popular) ? "Plan EstÃ¡ndar >" : "Plan Premiun >"}
                        </div>
                    </div>
                </div>
            </div>
            <div
                className="datos__modal"
                style={(modal) ? {display: 'none'}: {}}
                hidden={!dimensiones}
            >
                <SelectPlans
                    dimensiones={dimensiones}
                    handleClose={handleModal}
                />
            </div>
        </div>
    )
}
