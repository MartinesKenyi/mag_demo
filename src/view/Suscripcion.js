import React from 'react'
import { Plans } from '../components/Plans'

export const Suscripcion = ({history}) => {
    return (
        <div className="background__body">

            <div className="filter">
                <div className="suscription__container_title">
                    <h1>Mag <span>plans</span> </h1>
                    <p>Elija un plan que funcione mejor para usted.</p>
                </div>

                <div className="suscription__plans">
                    <Plans
                        history={history}
                    />
                </div>
            </div>
        </div>
    )
}
