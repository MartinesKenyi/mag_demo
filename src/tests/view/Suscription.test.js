import React from 'react';
import { shallow } from 'enzyme';
import { Suscripcion } from '../../view/Suscripcion';

describe('Pruebas en <Suscription />', () => {
   
    const historyMock = {
        push: jest.fn()
    }

    const wrapper = shallow(
        <Suscripcion 
            history={historyMock}
        />
    )

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    
});
