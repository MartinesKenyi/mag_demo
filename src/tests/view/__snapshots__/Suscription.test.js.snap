// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Suscription /> Debe de mostrarse correctamente 1`] = `
<div
  className="background__body"
>
  <div
    className="filter"
  >
    <div
      className="suscription__container_title"
    >
      <h1>
        Mag 
        <span>
          plans
        </span>
         
      </h1>
      <p>
        Elija un plan que funcione mejor para usted.
      </p>
    </div>
    <div
      className="suscription__plans"
    >
      <Plans
        beneficios={
          Array [
            "Sed ut perspiciatis unde omnis iste natus. ",
            "But I must explain to you how all this mistaken. ",
            "Sed ut perspiciatis unde omnis iste natus error.",
          ]
        }
        descricion="Lorem ipsum dolor sit amet, consectetur adipiscing eli"
        history={
          Object {
            "push": [MockFunction],
          }
        }
        key="0"
        popular={true}
        precio="29"
        tipo="EstÃ¡ndar"
      />
      <Plans
        beneficios={
          Array [
            "Sed ut perspiciatis unde omnis iste natus. ",
            "But I must explain to you how all this mistaken. ",
            "Sed ut perspiciatis unde omnis iste natus error.",
            "But I must explain to you how all this mistaken. ",
            "Sed ut perspiciatis unde omnis iste natus error.",
          ]
        }
        descricion="Lorem ipsum dolor sit amet, consectetur adipiscing eli"
        history={
          Object {
            "push": [MockFunction],
          }
        }
        key="1"
        popular={false}
        precio="59"
        tipo="Premiun"
      />
    </div>
  </div>
</div>
`;
